.github/workflows/merge-into-main.yml

```diff
@@ -0,0 +1,48 @@
+name: 'Merge Into Main'
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  terraform-changes:
+    name: Get All Top Level Directories Changed Under `terraform/`
+    outputs:
+      directories: ${{ steps.changed-terraform-directories.outputs.matrix }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: git fetch origin main:refs/remotes/origin/main
+      - id: changed-terraform-directories
+        run: |
+          top_level_directories_changed=$(git diff HEAD^1 --name-only -- terraform/   `# only compare files changed under the "terraform/" directory` \
```

Comment:
changed from `orign/main` to `HEAD^1` since the main branch won't be able to compare itself to get a diff.

Using `HEAD^1` restricts us to only using squash merges in Pull Requests. There are probably ways round this, but I believe this is "good enough" for an initial run.

We'll need to do some refactoring around the actions (probably a separate repo) if our projects start using the template 😄